source/extensions/load_balancing_policies/common/load_balancer_impl.cc

```diff
@@ -664,13 +692,17 @@ ZoneAwareLoadBalancerBase::calculateLocalityPercentages(
     }
     const auto& locality = upstream_hosts[0]->locality();
 
-    const auto& local_count_it = local_counts.find(locality);
-    const uint64_t local_count = local_count_it == local_counts.end() ? 0 : local_count_it->second;
+    const auto& local_weight_it = local_weights.find(locality);
+    const uint64_t local_weight =
+        local_weight_it == local_weights.end() ? 0 : local_weight_it->second;
+    const auto& upstream_weight_it = upstream_weights.find(locality);
+    const uint64_t upstream_weight =
+        upstream_weight_it == upstream_weights.end() ? 0 : upstream_weight_it->second;
```

Comment:
nit: iterator needn't to be reference. So, `const auto local_weight_iter` and `const auto upstream_weight_it` is better.